### ISSUE TITLE
OS X runtime error

### ISSUE THREAD

**Issue by bobakker** [tomaka/glium#257]
Glium finally compiles on OS X, but I'm getting the following runtime error:

thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: IncompatibleOpenGl("OpenGL implementation doesn\'t support vertex array objects\nOpenGL implementation doesn\'t support sampler objects")', /Users/rustbuild/src/rust-buildbot/slave/beta-dist-rustc-mac/build/src/libcore/result.rs:746

I'm pretty sure it's not an error in my code...

```rust
extern crate glutin;
extern crate glium;

use glium::DisplayBuild;

fn main() {
	let display = glutin::WindowBuilder::new()
		.with_gl_version((4, 1))				// If you're wondering, this isn't it.
		.build_glium()
		.unwrap();

	'main: loop {
		for event in display.poll_events().into_iter() {
			match event {
				glutin::Event::Closed => break 'main,
				_ => ()
			}
		}
	}
}
```

**Comment by tomaka** [comment]
Unfortunately there are many issues when creating an OpenGL context on OS/X with glutin :-/
It's probably related to https://github.com/tomaka/glutin/issues/176, although I'm not totally sure.

**Comment by bobakker** [comment]
Doesn't this fork fix some of these issues?
https://github.com/tomaka/glutin/pull/177

**Comment by tomaka** [comment]
I don't know. The author claims that it doesn't work.
Unfortunately the people who wrote the osx implementation of glutin have been busy lately, so the code has been neglected :-/ And I can't help it because I don't have any mac to try if things work.

**Comment by bobakker** [comment]
I'd love to help out, but I don't think I know enough about cocoa to be of much help. I'll be sure to look through the code to see if I can find anything, though.

**Comment by Nemo157** [comment]
Appears to be fixed now with tomaka/glutin#208, adding `.with_gl_version((4, 1))` to the window builder in the [complete example](http://tomaka.github.io/glium/glium/index.html#complete-example) and changing the shaders to be 4.10 compatible has got me a nice multi-coloured triangle. Next issue seems to be a complete lack of events :frowning:.
